27_tictactoe/main.py

```diff
@@ -0,0 +1,46 @@
+import re
+import lib.board as board
+from lib.player import Player
+
+players = [None,None]
+i = 0
+
+while True:
+
+    # Get players' info
+    while i < 2:
+        try:
+            players[i] = Player(name   = raw_input("Player %d - Name:   " %(i)),
+                                marker = raw_input("Player %d - Marker: " %(i)))
+            i += 1
+        except Exception as e:
+            print "Error occured while setting player %d\nError message: %s" %(i, e.message)
+
+
+    # Init game
+    tto = board.Game()
+    tto.start(players[0], players[1])
+
+    # Play game
+    while (tto.winner is None) and (not tto.tie()):
+        current_play = raw_input("%s's play (enter row,col): " % tto.current_player.name)
+        m = re.search(r'^\s*(\w)[,;\s\.]*(\w)\s*$', current_play)
+        try:
+            tto.play(int(m.group(1)),int(m.group(2)))
+            print tto
+        except board.InvalidMove as e:
+            print e.message
+        except Exception as e:
+            print e.message
+
+    # Print results
+    if tto.tie(): print "Game tied!!"
+    else: print "Winner is %s - well done!!\n" %(tto.winner.name)
+    print "------- GAME OVER --------\n"
+
+    # Continue?
+    if raw_input("Would you like to play again (y/n)? ").lower() == 'n': 
+        print "\nGood bye... Hope you enjoyed!!\n"
+        break
+    if raw_input("Same players (y/n)? ").lower() == 'n': 
+        i = 0
```

Comment:
All tests passed for me last night and still pass today…
Please advise.

[cid:image001.png@01D1E978.D48C8020]


From: ynonp [mailto:notifications@github.com]
Sent: Friday, July 29, 2016 8:38 AM
To: ynonp/python-examples-verint-2016-07
Cc: Naor, Nir; Author
Subject: Re: [ynonp/python-examples-verint-2016-07] ToCode Solution: תרגיל מסכם מחלקות (#221)


In 27_tictactoe/main.py<https://github.com/ynonp/python-examples-verint-2016-07/pull/221#discussion_r72743639>:

> +        except board.InvalidMove as e:

> +            print e.message

> +        except Exception as e:

> +            print e.message

> +

> +    # Print results

> +    if tto.tie(): print "Game tied!!"

> +    else: print "Winner is %s - well done!!\n" %(tto.winner.name)

> +    print "------- GAME OVER --------\n"

> +

> +    # Continue?

> +    if raw_input("Would you like to play again (y/n)? ").lower() == 'n':

> +        print "\nGood bye... Hope you enjoyed!!\n"

> +        break

> +    if raw_input("Same players (y/n)? ").lower() == 'n':

> +        i = 0

General Comments:

1.      Unit tests don't pass (run with python -m unittest discover)

2.      Gameplay is fun! Error handling works well

3.      Extra Points for the docstring documentation (however docstring for methods should appear as the first lines inside the method, not before it)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ynonp/python-examples-verint-2016-07/pull/221/files/e7ea17bf98533b25fcd24aa58240127dd7112296#r72743639>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ATWlusWShmUteXyks3o-K6wb6QCSJqyIks5qaZHMgaJpZM4JXyZi>.

-- 
This electronic message may contain proprietary and confidential 
information of Verint Systems Inc., its affiliates and/or subsidiaries.
The information is intended to be for the use of the individual(s) or
entity(ies) named above. If you are not the intended recipient (or 
authorized to receive this e-mail for the intended recipient), you may not 
use, copy, disclose or distribute to anyone this message or any information 
contained in this message. If you have received this electronic message in 
error, please notify us by replying to this e-mail.